sway-lib-std/tests/test_artifacts/reentrancy_attacker_contract/src/main.sw

```diff
@@ -0,0 +1,78 @@
+contract;
+
+use std::{chain::auth::*, constants::ZERO, context::call_frames::contract_id, contract_id::ContractId, panic::panic, result::*};
+use reentrancy_target_abi::Target;
+use reentrancy_attacker_abi::Attacker;
+
+// Return the sender as a ContractId or panic:
+fn get_msg_sender_id_or_panic(result: Result<Sender, AuthError>) -> ContractId {
+    let mut ret = ~ContractId::from(ZERO);
+    if result.is_err() {
+        panic(0);
+    } else {
+        let unwrapped = result.unwrap();
+        if let Sender::ContractId(v) = unwrapped {
+            ret = v;
+        } else {
+            panic(0);
+        };
+    };
+
+    ret
+}
+
+impl Attacker for Contract {
+    fn launch_attack(target: ContractId) -> bool {
+        let id = target.value;
+        let target = abi(Target, id);
+        target.reentrancy_detected()
+    }
+
+    fn launch_thwarted_attack_1(target: ContractId) {
+        let id = target.value;
+        let target = abi(Target, id);
+        target.reentrance_denied();
+    }
+
+    fn launch_thwarted_attack_2(target: ContractId) {
+        let id = target.value;
+        let target = abi(Target, id);
+        target.cross_function_reentrance_denied();
+    }
+
+    fn innocent_call(target: ContractId) -> bool {
+        let id = target.value;
+        let target = abi(Target, id);
+        let return_value = target.guarded_function_is_callable();
+        return_value
```

Comment:
```suggestion
        target.guarded_function_is_callable()
```

Reply:
I'll try to clean the tests up, they drifted a bit while just make it work.
This is what I had originally. I remember assigning to a variable to silence a compiler warning, but I should be able to make this better.

Reply:
cleaned up.